include/dgl/runtime/ndarray.h

```diff
@@ -272,6 +291,27 @@ class NDArray {
   DGL_DLL static void CopyFromTo(
       DLTensor* from, DLTensor* to, DGLStreamHandle stream = nullptr);
 
+  /*!
+   * \brief Function to pin the data.
+   * \param tensor The array to be pinned.
+   * \note Data of the given array will be pinned inplace.
+   *       Behavior depends on the current context,
+   *       kDLCPU: will be pinned;
+   *       kDLCPUPinned: directly return;
+   *       kDLGPU: invalid, will throw an error.
+   */
+  DGL_DLL static void PinData(DLTensor* tensor);
```

Comment:
What's the difference between `pinmemory` and `pindata`?
I prefer to use the same word here. Otherwise it confused me.

Reply:
`PinData` is a static method to pin the underlying DLTensor of an `NDArray`, while `PinMemory_()` is an inplace method to pin the current `NDArray` by calling `PinData`. `PinData` was previously introduced as a device API by David Min for UnifiedTensor, which directly manipulates the DGLArray. So I just follow it.

Reply:
I don't mind of renaming it! Current naming does not have a special meaning.

Reply:
I prefer to keep `PinData`. `PinMemory` (without underline) could be reserved for future out-of-place pinning for async transfer.

Reply:
Update comments to clarify their difference.